### ISSUE TITLE
Implement beta family

### ISSUE THREAD

**Issue by tomicapretto** [bambinos/bambi#310]
As far as I know, Bambi lacks support for beta regression since we only support `gaussian`, `bernoulli`, `poisson` and `t` as family names. I think we should implement a `beta` family to enable beta regression models.

**UPDATE** 

I was reading an outdated part of the docs. Current supported families are  `gaussian`, `bernoulli`, `poisson`, `gamma`, `wald` and `negativebinomial`. We still lack of `beta`.

**Comment by aloctavodia** [comment]
Currently our family support is restricted by statsmodels, which still lack beta regression statsmodels/statsmodels#4238. Is this a proposal to fix this upstream, fix it within Bambi, or move away from statsmodels? An the same questions for #311

**Comment by tomicapretto** [comment]
I hadn't think about what was the best alternative when opening the issue, but I think we should do whatever is easiest. 

In the long term, I would like to move away from statsmodels, so implementing these two custom families may me a good start. However, i don't know how difficult it is going to be.

If we see it is too difficult to use both our code and statsmodels implementations, we could consider contributing to statsmodels. However that implies learning about a new code base and it won't be that fast.

To summarize, I would go for our own implementation and try to use that as a first step in moving away from statsmodels. I could work on that. 

If I see that does not work for any reason, I'll update here

**Comment by aloctavodia** [comment]
This gist could work as a template for an intermediate solution https://gist.github.com/brentp/089c7d6d69d78d26437f 

It is for beta regression, but it could work as a template for student t regression.

**Comment by tomicapretto** [comment]
For the record and adding to Osvaldo's comment, [this](https://github.com/statsmodels/statsmodels/pull/7282) is the PR where the Beta family is being implemented in statsmodels.
